asyncapi/asgardeo/data_types.bal

```diff
@@ -141,6 +151,26 @@ public type LoginSuccessData record {
     string serviceProvider?;
     string userName?;
     string userId?;
+    AuthStep[] authSteps?;
+};
+
+# Login Failed Data
+#
+# + organizationId - Organization Id  
+# + ref - Reference  
+# + organizationName - Organization Name  
+# + serviceProvider - Service provider  
+# + authenticatingUser - User Name  
+# + userId - User Id
+# + failedStep -  Authentication failed step
+public type LoginFailedData record {
+    int organizationId?;
+    string ref?;
+    string organizationName?;
+    string serviceProvider?;
+    string authenticatingUser?;
+    string userId?;
+    AuthStep failedStep?;
```

Comment:
In this record all the fields are optional, does this mean that when there is a login failed event all of these fields are nilable ?

Reply:
Only field which has a chance of being null is userID.

Reply:
As @ayeshLK mentioned you can avoid marking them optional if they are mandatory. and if the userId can be `nil` you can mark it as billable. 

```suggestion
    int organizationId;
    string ref;
    string organizationName;
    string serviceProvider;
    string authenticatingUser;
    string? userId;
    AuthStep failedStep;
```

Reply:
Please check other places that @ayeshLK mentioned.

Reply:
Resolved in Resolved in [0dc5359](https://github.com/ballerina-platform/asyncapi-triggers/pull/158/commits/0dc53596be796d9198c0557bd0a1037a6fde9d10)